test/e2e_validate_data/test_validate_data.py

```diff
@@ -0,0 +1,176 @@
+from functools import lru_cache
+from pathlib import Path
+from typing import Iterator
+
+import pytest
+
+from dsp_tools.cli.args import ServerCredentials
+from dsp_tools.commands.project.create.project_create import create_project
+from dsp_tools.commands.validate_data.models.input_problems import ContentRegexViolation
+from dsp_tools.commands.validate_data.models.input_problems import MaxCardinalityViolation
+from dsp_tools.commands.validate_data.models.input_problems import MinCardinalityViolation
+from dsp_tools.commands.validate_data.models.input_problems import NonExistentCardinalityViolation
+from dsp_tools.commands.validate_data.models.input_problems import ValueTypeViolation
+from dsp_tools.commands.validate_data.models.validation import ValidationReports
+from dsp_tools.commands.validate_data.reformat_validaton_result import reformat_validation_graph
+from dsp_tools.commands.validate_data.validate_data import _get_validation_result
+from test.e2e_validate_data.setup_testcontainers import get_containers
+
+CREDS = ServerCredentials("root@example.com", "test", "http://0.0.0.0:3333")
+LOCAL_API = "http://0.0.0.0:3333"
+SAVE_GRAPHS = False
+
+
+@pytest.fixture
+def _create_project() -> Iterator[None]:
+    with get_containers():
+        success = create_project(Path("testdata/validate-data/project.json"), CREDS, verbose=True)
+        assert success
+        yield
+
+
+@lru_cache(maxsize=None)
```

Comment:
It is quite slow so I tried to solve this with a cache, please comment if you have any other ideas

Reply:
I am not familiar enough with python testcontainers and pytest fixtures to give a hint here. 

Currently all validation reports in these fixtures really do not depend on the state of the api so caching them seems fine to me.